include/apple2_filetype.h

```diff
@@ -0,0 +1,322 @@
+#define PRODOS_AUX_T_MLR_CHARACTER_DEF  0xCDEF  /* Mika City character def  */
+
+/* LDF auxiliary types */
+#define PRODOS_AUX_T_LDF_NIFTY_LIST_MOD 0x4001  /* Nifty list module        */
+#define PRODOS_AUX_T_LDF_SUPER_INFO_MOD 0x4002  /* Super info module        */
+#define PRODOS_AUX_T_LDF_TWILIGHT_MOD   0x4004  /* Twilight module          */
+#define PRODOS_AUX_T_LDF_MARINETTI_LLM  0x4004  /* Marinetti link layer mod */
+
+/* PNT auxiliary types */
+#define PRODOS_AUX_T_PNT_PK_SUPER_HIRES 0x0001  /* Packed super hi-res      */
+#define PRODOS_AUX_T_PNT_APPLE_PREF     0x0002  /* Apple preferred format   */
+#define PRODOS_AUX_T_PNT_PK_QUICKDRAWII 0x0003  /* Packed QuickDraw II      */
+
+/* PIC auxiliary types */
+#define PRODOS_AUX_T_PIC_QUICKDRAW      0x0001  /* QuickDraw image          */
+#define PRODOS_AUX_T_PIC_SHIRES_3200    0x0002  /* Super hi-res 3200        */
+
+/* FON auxiliary types */
+#define PRODOS_AUX_T_FON_QUICKDRAW_BIT  0x0000  /* QuickDraw bitmap font    */
+#define PRODOS_AUX_T_FON_POINTLESS_TT   0x0001  /* Pointless TrueType font  */
+
+/* SND auxiliary types */
+#define PRODOS_AUX_T_SND_AIFF           0x0000  /* AIFF                     */
+#define PRODOS_AUX_T_SND_AIFF_C         0x0001  /* AIFF-C                   */
+#define PRODOS_AUX_T_SND_ASIF_INSTR     0x0002  /* ASIF instrument          */
+#define PRODOS_AUX_T_SND_SOUND_RSRC     0x0003  /* Sound resource           */
+#define PRODOS_AUX_T_SND_MIDI_SYNTH_WAV 0x0004  /* MIDI synth wave          */
+#define PRODOS_AUX_T_SND_HYPERSTUDIO    0x8001  /* HyperStudio sound        */
+
+/* LBR auxiliary types */
+#define PRODOS_AUX_T_LBR_ALU            0x0000  /* ALU                      */
+#define PRODOS_AUX_T_LBR_APPLE_SINGLE   0x0001  /* AppleSingle              */
+#define PRODOS_AUX_T_LBR_APPLEDBL_HDR   0x0002  /* AppleDouble header       */
+#define PRODOS_AUX_T_LBR_APPLEDBL_DATA  0x0003  /* AppleDouble data         */
+#define PRODOS_AUX_T_LBR_BINARY_II      0x8000  /* Binary II                */
+#define PRODOS_AUX_T_LBR_APPLELINK_ACU  0x8001  /* AppleLink ACU            */
+#define PRODOS_AUX_T_LBR_SHRINKIT       0x8002  /* ShrinkIt                 */
+
+/* LBR auxiliary types */
+#define PRODOS_AUX_T_ATK_EASYMNT_ALIAS  0x0000  /* EasyMount alias          */
+
+/* BAS auxiliary types */
+#define PRODOS_AUX_T_BAS_PGM_LOAD_ADDR  0x0801  /* Applesoft pgm load addr  */
+
+
+
+/*****************************************************************************/
+/*                                 Variables                                 */
+/*****************************************************************************/
+
+
+
+/* The file stream implementation and the POSIX I/O functions will use the
+** following variables to determine the file type and auxiliary type to use.
+** This applies specifically to the fopen and open functions.
+*/
+extern unsigned char _filetype;  /* Default: PRODOS_T_BIN */
+extern unsigned int _auxtype;    /* Default: 0 */
```

Comment:
As you did so nicely in the rest of this file you might want to align the closing comments here too.